src/c++/perf_analyzer/command_line_parser.cc

```diff
@@ -717,6 +723,13 @@ CLParser::Usage(const std::string& msg)
   exit(GENERIC_ERROR);
 }
 
+void
+CLParser::PrintVersion()
+{
+  std::cerr << "Perf Analyzer Version: " << VERSION << std::endl;
```

Comment:
It would be really cool to include the git SHA in the version. Docker does something similar to this:

```
$ docker --version
Docker version 24.0.2, build cb74dfc
```

https://github.com/docker/cli/commits/24.0
https://github.com/docker/cli/commit/cb74dfcd853482dd43cb553106b1e0cd237acb3e

@mc-nv for ideas on how to accomplish this. I can imagine it would involve setting some define using cmake, and then the c++ code just uses the define.

Reply:
that is possible with a cmake configure file and some scripting. 
i think this falls out of scope of the original ticket but I will look into it to see if its straightforward

Reply:
Sorry @matthewkotila, pushing back on this one.
I tried for a couple hours and im getting this error: 
```fatal: detected dubious ownership in repository at '/workspace/client'
To add an exception for this directory, call:

  git config --global --add safe.directory /workspace/client
-- GIT SHA RC: 128
-- GIT SHA:
```

Basically, I would have to make the .git directory safe, which removes security protections, to be able to run this command. 
I think the way forward is to pass this in via docker build_args to make sure its available in the container and set correctly.
Alternatively, we can create a cmake .in file with the build sha. Either way, this is out of scope for the current ticket. 
I am going to create a new ticket to track this work.

Reply:
The reason you're running into the error is because you're running cmake from an environment where the current user is different from the user that is the owner (`chown ...`) of the repository directories/files.

It's perfectly fine to run the suggested command (`git config --global --add safe.directory /workspace/client`) from your development environment. When the actual CI runs to build Perf Analyzer, the user running cmake will be the same user that owns the repository directory/files.

Reply:
Updated to get the sha into the version output.